source/common/grpc/grpc_web_filter.cc

```diff
@@ -87,6 +90,13 @@ Http::FilterDataStatus GrpcWebFilter::decodeData(Buffer::Instance& data, bool) {
   const std::string decoded = Base64::decode(
       std::string(static_cast<const char*>(decoding_buffer_.linearize(decoding_buffer_.length())),
                   decoding_buffer_.length()));
+  if (decoded == EMPTY_STRING) {
+    // Error happened when decoding base64.
+    Http::Utility::sendLocalReply(*decoder_callbacks_, stream_destroyed_, Http::Code::BadRequest,
```

Comment:
It's not standardized yet. We do:
1. return what makes sense to HTTP client if the request get rejected by the proxy.
2. return whatever we received from gRPC backend if the request was sent to the backend.